File: Bachelor Thesis/backend/src/main/kotlin/ro/ubb/flaviu/models/moves/Move.kt
package ro.ubb.flaviu.models.moves

import ro.ubb.flaviu.models.Color
import ro.ubb.flaviu.models.Position

interface Move {
    val initialPosition: Position
    val finalPosition: Position
    val color: Color
    val isCapture: Boolean
    val score: Double
}

File: Bachelor Thesis/frontend/src/app/models/move.ts
import {Position} from "./position";

export class Move {
  initialPosition: Position
  finalPosition: Position
  color: String
  className: string


  constructor(initialPosition: Position, finalPosition: Position, color: String, className: string) {
    this.initialPosition = initialPosition;
    this.finalPosition = finalPosition;
    this.color = color;
    this.className = className;
  }
}
